src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -392,7 +394,22 @@ private function dehydrateObjectValue(object $value, string $classType, ?string
             }
         }
 
-        throw new \LogicException(sprintf('Unable to dehydrate value of type "%s" for property "%s" on component "%s". Either (1) change this to a simpler value, (2) add the hydrateWith/dehydrateWith options to LiveProp or (3) set "useSerializerForHydration: true" on the LiveProp.', $value::class, $propertyPathForError, $componentClassForError));
+        $reflexionExtractor = new ReflectionExtractor();
+        $properties = $reflexionExtractor->getProperties($classType);
+        $propertiesValues = [];
+        foreach ($properties as $property) {
+            if ($reflexionExtractor->isReadable($classType, $property)) {
+                $propertyValue = $this->propertyAccessor->getValue($value, $property);
+                $type = $reflexionExtractor->getTypes($classType, $property)[0]->getBuiltinType();
+                if ($type === 'object') {
+                    $type = $reflexionExtractor->getTypes($classType, $property)[0]->getClassName();
+                }
+                $propMetadata = new LivePropMetadata($property, new LiveProp(true), $type, false, true, null);
+                $propertiesValues[$property] = $this->dehydrateValue($propertyValue, $propMetadata, $component);
+            }
+        }
```

Comment:
Yes, I think this is definitely what I had in mind!!! Details:

* A) What about support for collections / arrays? So, something like this should also be allowed to work in the DTO

```php
class CustomerDetails
{
    public string $name;
    /** @var Address[] */
    public array $addresses;
    public string $city;
}
```

This type of thing DOES work on the top-level component (e.g. you could have a `LiveProp` that is an array of entity objects). 

* B) Your extracting the "types" and properties using the `ReflectionExtractor`. Can we use the PropertyInfo component services instead? That's especially important for the types: we need to use the same logic for determining the types in the DTOs as we do on the top-level `LiveProp`s - https://github.com/symfony/ux/blob/2.x/src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php#L62-L84

Ideally, we would not only match that, but centralize it somehow: whenever we need a "type" for any property that we (de)hydrate, we use the same code.